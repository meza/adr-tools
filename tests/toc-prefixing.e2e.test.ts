/* eslint-disable no-sync */
import { describe, it, expect, afterEach, beforeEach, vi } from 'vitest';
import * as childProcess from 'child_process';
import * as path from 'path';
import * as fs from 'fs/promises';
import * as os from 'os';

describe('Generating TOC', () => {
  const adr = path.resolve(path.dirname(__filename), '../src/index.ts');
  const command = `npx tsx ${adr}`;

  let adrDirectory: string;
  let workDir: string;

  beforeEach(async () => {
    // @ts-ignore
    process.env.ADR_DATE = '1992-01-12';
    workDir = await fs.mkdtemp(path.join(os.tmpdir(), 'adr-'));
    adrDirectory = path.join(workDir, 'doc/adr');
  });

  afterEach(() => {
    vi.clearAllMocks();
    childProcess.execSync(`rm -rf ${workDir}`);
  });

  it('should add a path prefix to the toc when there is one supplied', async () => {
    childProcess.execSync(`${command} new First Record`, { cwd: workDir });
    childProcess.execSync(`${command} new Second Record`, { cwd: workDir });
    childProcess.execSync(`${command} new Third Record`, { cwd: workDir });
    childProcess.execSync(`${command} generate toc -p foo/doc/adr/`, { cwd: workDir });

    const tocFilePath: string = path.join(adrDirectory, 'decisions.md');
    const tocContent = await fs.readFile(tocFilePath, 'utf8');
    expect(tocContent).toMatchSnapshot();
  });

});
