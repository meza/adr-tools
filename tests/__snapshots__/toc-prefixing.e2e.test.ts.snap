// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Generating TOC > should add a path prefix to the toc when there is one supplied 1`] = `
"# Table of Contents

- [1. First Record](foo/doc/adr/0001-first-record.md)
- [2. Second Record](foo/doc/adr/0002-second-record.md)
- [3. Third Record](foo/doc/adr/0003-third-record.md)"
`;
